packages/flutter/test/material/dialog_theme_test.dart

```diff
@@ -548,4 +550,54 @@ void main() {
       ),
     );
   });
+
+  testWidgets('DialogTheme.clipBehavior updates the dialogs clip behavior', (WidgetTester tester) async {
+    const DialogTheme dialogTheme = DialogTheme(
+      clipBehavior: Clip.hardEdge
+    );
+    const Dialog dialog = Dialog(child: Placeholder());
+
+    await tester.pumpWidget(_appWithDialog(
+      tester,
+      dialog,
+      theme: ThemeData(dialogTheme: dialogTheme),
+    ));
+
+    final Material materialWidget = _getMaterialFromDialog(tester);
```

Comment:
```suggestion
    final Material materialWidget = _getMaterialFromDialog(tester);
```

Tests are failing due to `_getMaterialFromDialog` finding `AlertDialog` not the `Dialog` widget.

Locate the this function and rename it and add another function for the `Dialog` like this:

```diff
+ Material _getMaterialAlertDialog(WidgetTester tester) {
  return tester.widget<Material>(find.descendant(of: find.byType(AlertDialog), matching: find.byType(Material)));
}
+ 
+ Material _getMaterialDialog(WidgetTester tester) {
+  return tester.widget<Material>(find.descendant(of: find.byType(Dialog), matching: find.byType(Material)));
+}
```